### DISCUSSION TITLE
SPDM 1.2 - GET_CERTIFICATE responder handling

### DISCUSSION THREAD

**Discussion by RohitPatil555** [DMTF/libspdm#755] [General]
Hi,

I saw code around get_certificate handling in responder. In that, I can see offset and length parameters got used.
But as per spec 1.2, responder will not receive offset and length field in GET_CERTIFICATE request message.

Is this Issue in code ?

If adding support for GET_CERTIFICATE v1.2 is in process then please share issue link with me.
I am interested to see.

Thanks,
~Rohit

**Comment by steven-bellock** [comment]
libspdm contains a lot of the infrastructure for SPDM 1.2 but we haven't enabled it yet. That work will begin in April.

**Reply by RohitPatil555** [reply]
Hi, 
What you mean as infrastructure ? Is it SPDM 1.2 features ?
or libSPDM basic framework ?

As I can see CHUNK support needed before implementing GET_CERTIFICATE as per 1.2.

**Reply by steven-bellock** [reply]
> Hi, What you mean as infrastructure ? Is it SPDM 1.2 features ? or libSPDM basic framework ?

As in https://github.com/DMTF/libspdm/blob/main/include/industry_standard/spdm.h and other files have most of the #defines and typedefs for SPDM 1.2. The work to enable the 1.2 request / response messages will begin in April.
